src/ProblemDocument.ts

```diff
@@ -61,9 +65,9 @@ export class ProblemDocumentOptions {
 }
 
 export class ProblemDocumentExtension {
-  public extensionProperties: any;
+  public extensionProperties: Record<string, any>;
 
-  public constructor (extensionProperties: any) {
+  public constructor (extensionProperties: Record<string, any>) {
```

Comment:
This is almost a breaking change already, but I think it's a fair assumption that only JS object would have been used and not for example `new ProblemDocumentExtension(1234)`

Reply:
`ProblemDocumentExtension` has been added to have a wording similar to the RFC - but I get the point.

How about `Record<string, string | number | Date | boolean | object>` to avoid `any`?

Reply:
Hm, on second thought plain `Object` does the trick...

Reply:
Sorry for the late response... So should we stick with `Record<string, object>`?

Reply:
I actually tried as 

```suggestion
  public constructor (extensionProperties: Record<string, Object>) {
```